### PR TITLE
Fix deleting properties after deleting files

### LINKED ISSUE #26929: Deleting files doesn't update some project settings properties
**Godot version:**
Godot 3.1 RC 1

**OS/device including version:**
Linux Mint 19

**Issue description:**
When I delete some files which are used by some of project settings properties, then project settings still have info about this files, which can cause some unexpected errors(probably just printed to output error)

Issue is same as #24377 but I think that all properties should be updated after file deletion.

### PR DESCRIPTION
**Pull request by qarmin**
This PR fix all project properties which can point to non existent files and show errors.

Fix #26929

### CODE REVIEW

File: editor/dependency_editor.cpp [hunk 1]
```diff
@@ -496,13 +496,21 @@ void DependencyRemoveDialog::ok_pressed() {
 			res->set_path("");
 		}
 
-		// If the file we are deleting is the main scene or default environment, clear its definition.
-		if (files_to_delete[i] == ProjectSettings::get_singleton()->get("application/run/main_scene")) {
+		// If the file we are deleting for e.g. the main scene or default environment, we must clear its definition in Project Settings.
+		if (files_to_delete[i] == ProjectSettings::get_singleton()->get("application/config/icon")) {
+			ProjectSettings::get_singleton()->set("application/config/icon", "");
+		} else if (files_to_delete[i] == ProjectSettings::get_singleton()->get("application/run/main_scene")) {
```
Comment (akien-mga) [inline editor/dependency_editor.cpp]: Should be a plain `if` and not `else if`, as some files might be used by several properties (e.g. icon, cursor and boot splash could be the same).

**Comment by reduz** [comment]
this is definitely useful

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.1.1.

**Pull request merged**